### PR TITLE
Code: Tweak docstrings and return type hints

### PR DESCRIPTION
**Pull request by BigRoy**
## Changelog Description

Tweak docstrings and return type hints for functions in `openpype.client.entities`.

## Additional info

As [reported on discord](https://discord.com/channels/517362899170230292/517382145552154634/1098511261291003966) this intends to solve the issue I had:

> I'm having some issues with many functions in the `openpype.client.entities` package. My IDE keeps telling me functions like these return `None` instead of `dict` or `None.` For example for `get_representation_by_name` https://github.com/ynput/OpenPype/blob/9aa8aa469fc3a81e714207809af786b520043bf6/openpype/client/entities.py#L1004-L1020

## Testing notes:

1. Launch your IDE with the codebase and hope it likes it.
2. Make sure you can read the docstrings and the are correct.

### CODE REVIEW

File: openpype/client/entities.py [hunk 1]
```diff
@@ -81,6 +94,19 @@ def get_projects(active=True, inactive=False, fields=None):
 
 
 def get_project(project_name, active=True, inactive=True, fields=None):
+    """Return project entity document by project name.
+
+    Args:
```
Comment (iLLiCiTiT) [inline openpype/client/entities.py:99]:
```suggestion
    Args:
        project_name (str): Name of project.
```

File: openpype/client/entities.py [hunk 2]
```diff
@@ -81,6 +94,20 @@ def get_projects(active=True, inactive=False, fields=None):
...
+        project_name (str): Name of project.
+        active (Optional[bool]): Allow active project. Defaults to True.
+        inactive (Optional[bool]): Allow inactive project. Defaults to True.
+        fields (Iterable[str]): Fields that should be returned. All fields are
+            returned if 'None' is passed.
```
Comment (BigRoy) [inline openpype/client/entities.py:104]: Should this be `Optional[Iterable[str]]` @iLLiCiTiT 

```suggestion
        fields (Optional[Iterable[str]]): Fields that should be returned.
            All fields are returned if 'None' is passed.
```

Comment (iLLiCiTiT) [reply openpype/client/entities.py:104]: Yes

Comment (BigRoy) [reply openpype/client/entities.py]: Done.

**Review by iLLiCiTiT — APPROVED** [review]

**Pull request merged**